### DISCUSSION TITLE
Is there a way to handle cancelled requests

### DISCUSSION THREAD

**Discussion by hbatalhaStch** [adonisjs/core#4408] [Help] [answered]
I am making requests to my AdonisJs api from a Flutter application using [dio](https://github.com/cfug/dio) and one of dio's features is the ability to [cancel a request](https://github.com/cfug/dio/tree/main/dio#cancellation).

After some tests I noticed that request cancellations done by the client has no effect on the api since it had already received the requests.

So I am wondering if it's possible to handle cancelled requests.

I'm using V5.

**Comment by muco-rolle** [comment]
Hey @hbatalhaStch does the cancellation send back any information to the backend? It does something to notify the server you should be able to handle that

**Reply by hbatalhaStch** [reply]
> does the cancellation send back any information to the backend?

no, it just kills that request connection to the backend.

in express we have something like this:
```js
authRouter.post("/api/signup", async (req, res) => {

    let requestCancelled = false;
    req.socket.on('close', async () => {
        requestCancelled = true;
    })
// rest of the code
}
```
we can use `requestCancelled` to opt out of any operation in the rest of the code and just return an error for example.

**Comment by holoyan** [comment, marked as answer]
```

router.get('/t', async (ctx) => {
  ctx.request.request.socket.on('close', () => {
    // do something
  })
})


```

from [source code](https://github.com/adonisjs/http-server/blob/main/src/request.ts#L107) seems that `ctx.request.request` is instance of [incoming request](https://nodejs.org/api/http.html#class-httpincomingmessage)

@hbatalhaStch

**Reply by hbatalhaStch** [reply]
thank you
